--- src/libplanet/Planet.cpp	2004-07-23 03:05:40.000000000 +0200
+++ src/libplanet/Planet.cpp	2005-05-14 14:24:12.589748439 +0200
@@ -341,9 +341,9 @@
             alpha0_ = 268.05 - 0.009 * T2000_;
             delta0_ = 64.49 + 0.003 * T2000_;
           
-            // System III (magnetic field rotation)
-            nullMeridian0_ = 284.95;
-            wdot_ = 870.536;
+            // System II (GRS rotation)
+            nullMeridian0_ = 43.3;
+            wdot_ = 870.27003539;
 
             period_ = 4332.589;
 
